plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -282,7 +286,7 @@ var _ = Describe("validator", func() {
 			ctx = context.Background()
 			ctrl = gomock.NewController(GinkgoT())
 			auth = nil
-
+			test.WithFeatureGate(schedulerfeatures.FeatureGate, features.HAControlPlanes, true)
```

Comment:
```suggestion
			DeferCleanup(test.WithFeatureGate(schedulerfeatures.FeatureGate, features.HAControlPlanes, true))
```

Also wouldn't it be better to put this line in a `BeforeEach()` in the `Context("HA shoot scheduling checks on seed", func() { ....` block?

Reply:
Feature gate usage was not required and also did not make much sense for admission plugin as no one explicitly adds any feature gates for plugins today. I have removed that check entirely and also changed the unit tests to remove `test.WithFeatureGate`

Reply:
I guess it makes sense that the check for the HA feature gate is not really needed. 
But just a note that we still use the `SeedChange` feature gate here: https://github.com/gardener/gardener/blob/b01fd6084bb89a42a45ce7f76e20cbd6f0fd17b0/plugin/pkg/shoot/binding/admission.go#L177